python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -440,6 +444,13 @@ def core_dagster_event_sequence_for_step(
         if dagster_type.is_nothing:
             continue
 
+        if (
+            isinstance(step_input.source, FromStepOutput)
+            and step_input.source.output_dagster_type_is_nothing
```

Comment:
I believe we should be able to use the the step output handle on `FromStepOutput` to look-up the source object and derive this information instead of having to capture it this way

Reply:
ok will take a look. that'd be much nicer